### PR TITLE
feat: Dashboard tabs api endpoint

### PR DESCRIPTION
**Pull request by fisjac**
### SUMMARY

This PR implements a new property on the Dashboard model granting access to the tabs associated with a dashboard. A new API endpoint for accessing the associated dashboard tabs has been created at `/api/v1/dashboard/[id]/tabs`.

The payload of the api endpoint contains two properties: `all_tabs` which contains an array of all associated tabs, and `tab_tree` which contains a nested object representing the nested structure of tabs within the dashboard. tab_tree follows the data structure necessary for consumption by the Antd Tree and TreeSelect components on the frontend.

The primary use case for this implementation is for enabling the functionalities described in SIP: https://github.com/apache/superset/issues/26183.

The provided proof of concept provides an alternative approach, centralizing the business logic on the backend to process prior to serving to the client.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: tests/integration_tests/dashboards/api_tests.py [hunk 1]
```diff
@@ -869,6 +869,185 @@ def test_get_dashboards_created_by_me(self):
...
+                    },
+                ],
+            }
+        }
+        self.assertEqual(rv.status_code, 200)
+        self.assertEqual(response, expected_response)
+        db.session.delete(dashboard)
+        db.session.commit()
+
+    @pytest.mark.usefixtures("load_world_bank_dashboard_with_slices")
```
Comment (geido) [inline tests/integration_tests/dashboards/api_tests.py:1040]: Should we add other negative paths, such as testing for permissions or invalid schema?

File: superset/dashboards/api.py [hunk 2]
```diff
@@ -395,6 +399,64 @@ def get_datasets(self, id_or_slug: str) -> Response:
...
+        """
+        try:
+            tabs = DashboardDAO.get_tabs_for_dashboard(id_or_slug)
+            result = self.tab_schema.dump(tabs)
+            return self.response(200, result=result)
+
+        except (TypeError, ValueError) as err:
+            return self.response_400(
+                message=gettext(
+                    "Tab schema is invalid, caused by: %(error)s", error=str(err)
```
Comment (geido) [inline superset/dashboards/api.py:452]: I don't think we need to surface the specific error here in the response. We might want to log it instead.

Comment (fisjac) [reply superset/dashboards/api.py:452]: I borrowed this error handler from the `GET dashboard/<id>/datasets` endpoint, which throws when an invalid schema is detected. My thought was to ensure consistent validation behavior for similar endpoints.

File: tests/integration_tests/dashboards/api_tests.py (hunk 1 above)
Comment (fisjac) [reply tests/integration_tests/dashboards/api_tests.py:1040]: currently the only tests that explicitly check permissions for the dashboard api are for PUT and DELETE requests. My understanding is that the FlaskAppbuilder @protect decorator is doing most of the work for ensuring permissions for GET requests. Presumably the @protect decorator is being tested somewhere else within the codebase?

**Review by geido — APPROVED** [review]

**Pull request merged**